client/src/game/ui/Button2.ts

```diff
@@ -112,4 +114,22 @@ export default class Button2 extends Konva.Group {
   setMiddleText(text: string) {
     this.textElement2.text(text);
   }
+
+  setEnabled(enabled: boolean) {
```

Comment:
if you do this, enable must become a readonly field. Otherwise someone can change it outside of this method. You probably want to use a proper getter and setter

Reply:
you mean the enabled property should be private?

Reply:
```
class Test
{
    private _prop: string;
    public get prop() : string
    {
        return this._prop;
    }

    doSomething() {
        this._prop = 'I can set it!';
    }
}
```